ui/app/hooks/useTransactionDisplayData.js

```diff
@@ -56,20 +56,21 @@ export function useTransactionDisplayData (transactionGroup) {
   const knownTokens = useSelector(getTokens)
   const t = useI18nContext()
   const { initialTransaction, primaryTransaction } = transactionGroup
-  // initialTransaction contains the data we need to derive the primary purpose of this transaction group
-  const { transactionCategory } = initialTransaction
+  const transaction = isCancelledTransaction(primaryTransaction) ? initialTransaction : primaryTransaction || initialTransaction
```

Comment:
Last question I have for both you and @Gudahtt -- in what cases would the `initialTransaction` be set and the `primaryTransaction` not be set? Just the first transaction in a group?

Reply:
I think we always set the primaryTransactions? https://github.com/MetaMask/metamask-extension/blob/develop/ui/app/selectors/transactions.js#L253-L261